### PR TITLE
Fixup toolbelt to use exposed key ID.

### PR DESCRIPTION
**Pull request by emgee**
The asserts API now requires the key ID to find a public key and to sign
a new assertion.

Hijacks the commands' existing positional args to include the key ID by
using the syntax <name>/<keyid>.

### CODE REVIEW

File: asserts/toolbelt/main.go [hunk 1]
```diff
@@ -80,12 +81,12 @@ func (x *generateKey) Execute(args []string) error {
...
 }
 
-func findAuthFingerprint(authID string) (string, error) {
-	authPubKey, err := db.PublicKey(authID, "")
-	if err != nil {
-		return "", fmt.Errorf("failed to find signing key pair: %v", err)
+func findPublicKey(db *asserts.Database, id string) (asserts.PublicKey, error) {
+	parts := strings.SplitN(id, "/", 2)
+	if len(parts) != 2 {
+		return nil, errors.New("id should be name/key-id")
```
Comment (niemeyer) [inline asserts/toolbelt/main.go]: fmt.Errorf(`invalid public key specification: %q (expected "account-name/key-id")`, id)

**Comment by niemeyer** [comment]
Not sure if we had agreed about the convention for account+key-id, but looks good for the time being.

LGTM with just one trivial suggestion for the error message.

File: asserts/toolbelt/main.go (hunk 1 above)
Comment (emgee) [inline asserts/toolbelt/main.go]: Fixed, thanks.

**Comment by emgee** [comment]
I used your error message suggestion.

afaik, the toolbelt is a temporary tool that's likely to be replaced. I'm using it to generate test assertions so just did the easiest thing that also seemed sensible to get the key ids in.

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by pedronis** [comment]
+1

**Pull request merged**